### PR TITLE
bpo-45569: Change PYLONG_BITS_IN_DIGIT default to 30

### PR DESCRIPTION
**Pull request by mdickinson**
This PR changes the default for `PYLONG_BITS_IN_DIGIT` to 30, unconditionally. 15-bit digits can still be selected explicitly via `configure` (Unix-like) or in `pyconfig.h` (Windows).

https://bugs.python.org/issue45569

### CODE REVIEW

**Comment by mdickinson** [comment]
There were 5 buildbot failures, none of which appears to be directly related to this PR. Most of them seem to be due to an enum module refleak, which has since been corrected (see https://bugs.python.org/issue46301). The ARM64/macOS PR is failing due to test_ftplib changing the environment.

I'll re-do the buildbot run, in the hope of seeing the refleak-related failures disappear.

**Comment by mdickinson** [comment]
@tiran Would you have bandwidth to do a sanity check on this PR?

**Comment by mdickinson** [comment]
Hmm. I need to update the wording in the `configure` script help, too. I'll do that.

File: Misc/NEWS.d/next/Build/2022-01-09-11-24-54.bpo-45569.zCIENy.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The build now defaults to using 30-bit digits for Python integers. Previously
+either 15-bit or 30-bit digits would be selected, depending on the platform.
+15-bit digits may still be selected using the ``--enable-big-digits=15`` option
+to the ``configure`` script, or by defining ``PYLONG_BITS_IN_DIGIT`` in
+``pyconfig.h``.
```
Comment (tiran) [inline Misc/NEWS.d/next/Build/2022-01-09-11-24-54.bpo-45569.zCIENy.rst:5]: ``pyconfig.h`` is auto-generated and not user-editable. I suggest to remove the qualifier.

```suggestion
to the ``configure`` script, or by defining ``PYLONG_BITS_IN_DIGIT``.
```

Comment (mdickinson) [reply Misc/NEWS.d/next/Build/2022-01-09-11-24-54.bpo-45569.zCIENy.rst:5]: True; I was thinking of the Windows setup. Maybe replacing ``pyconfig.h`` with ``PC/pyconfig.h`` would be enough?

File: configure.ac [hunk 2]
```diff
@@ -5039,7 +5039,7 @@ AC_CHECK_DECLS([RTLD_LAZY, RTLD_NOW, RTLD_GLOBAL, RTLD_LOCAL, RTLD_NODELETE, RTL
 # determine what size digit to use for Python's longs
 AC_MSG_CHECKING([digit size for Python's longs])
 AC_ARG_ENABLE(big-digits,
-AS_HELP_STRING([--enable-big-digits@<:@=15|30@:>@],[use big digits (30 or 15 bits) for Python longs (default is system-dependent)]]),
```
Comment (tiran) [inline configure.ac:5042]: You have to update ``configure``, too. It's a trivial change, so manually editing is easier.

File: Misc/NEWS.d/next/Build/2022-01-09-11-24-54.bpo-45569.zCIENy.rst (hunk 1 above)
Comment (mdickinson) [reply Misc/NEWS.d/next/Build/2022-01-09-11-24-54.bpo-45569.zCIENy.rst:5]: OTOH, probably not worth worrying about explicitly specifying how to change things for Windows here.

File: configure.ac (hunk 2 above)
Comment (mdickinson) [reply configure.ac:5042]: Yes, I've spent the last hour trying to get `autoreconf` to not generate spurious changes. :-( I'll go for the manual edit option.

File: Misc/NEWS.d/next/Build/2022-01-09-11-24-54.bpo-45569.zCIENy.rst (hunk 1 above)
Comment (mdickinson) [reply Misc/NEWS.d/next/Build/2022-01-09-11-24-54.bpo-45569.zCIENy.rst:5]: Clarified (I hope).

File: configure.ac (hunk 2 above)
Comment (mdickinson) [reply configure.ac:5042]: Done.

Comment (tiran) [reply configure.ac:5042]: > Yes, I've spent the last hour trying to get `autoreconf` to not generate spurious changes. :-( I'll go for the manual edit option.

I have you covered: https://quay.io/repository/tiran/cpython_autoconf

**Review by tiran — APPROVED** [review]

**Comment by mdickinson** [comment]
Thanks, @tiran!

**Pull request merged**